client/keys/add.go

```diff
@@ -202,6 +210,27 @@ func runAddCmd(ctx client.Context, cmd *cobra.Command, args []string, inBuf *buf
 
 		return printCreate(ctx, cmd, k, false, "", outputFormat)
 	}
+	if pubKeyBase64 != "" {
+		b64, err := base64.StdEncoding.DecodeString(pubKeyBase64)
+		if err != nil {
+			return err
+		}
+		var pk cryptotypes.PubKey
+		// create an empty pubkey in order to get the algo TypeUrl.
+		tempAny, err := codectypes.NewAnyWithValue(algo.Generate()([]byte{}).PubKey())
+		if err != nil {
+			return err
+		}
+		jsonPub := fmt.Sprintf(`{"@type":"%s","key":"%s"}`, tempAny.TypeUrl, b64)
```

Comment:
Is there a way to avoid manual JSON templating?

Reply:
Yeah this is susceptible to some sneak attacks, we can invoke this
```go
jsonPubB, err := json.Marshal(struct{
     Type string `json:"@type,omitempty"`
     Key string `json:"key,omitempty"`
}{tempAny.TypeURL, b64})
if err != nil {
    return fmt.Errorf("failed to JSON marshal typeURL and base64 key: %w", err)
}
```

Reply:
nice catch guys!! Resolved in last commit!